storage/Devices/BlkDeviceImpl.cc

```diff
@@ -825,4 +867,46 @@ namespace storage
 	}
     }
 
+
+    void
+    wait_for_detach_devices(const vector<const BlkDevice*>& blk_devices)
+    {
+	vector<string> dev_names;
+
+	for_each(blk_devices.begin(), blk_devices.end(), [&dev_names](const BlkDevice* dev)
+	{
+	    dev_names.push_back(dev->get_name());
+	});
+
+	wait_for_detach_devices(dev_names);
+    }
+
+
+    void
+    wait_for_detach_devices(const vector<string>& dev_names)
+    {
+	SystemCmd(UDEVADMBIN_SETTLE);
+
+	for (auto name : dev_names)
+	{
+	    bool exists = access(name.c_str(), R_OK) == 0;
+	    y2mil("name:" << name << " exists:" << exists);
+
+	    if (exists)
+	    {
```

Comment:
Please add an info/milestone logging here so we know what's going on when we get bug report that YaST hangs. Since this might wait for quite a while, it will be unclear what's going on here, and it will be very hard to debug without such a logging line.